query-planner-js/src/FieldSet.ts

```diff
@@ -200,6 +200,10 @@ function mergeFieldNodeSelectionSets(
     (node): node is FieldNode => node.kind === Kind.FIELD,
   );
 
+  // Committed by @trevor-scheer but authored by @martijnwalraven
```

Comment:
I think @martijnwalraven might be interested in looking over these comments and making sure they fully capture what he wants, but that could be after this PR merges.